tests/ReduceTest.php

```diff
@@ -53,7 +53,7 @@ public function test_iteration_should_halt_early_when_given_signal() {
             return f\reduce(
                 function ($_, $item) use ($predicate) {
                     if ($predicate($item)) {
-                        return new f\HaltReductionSignal($item);
+                        return new f\ReducedValue($item);
```

Comment:
Maybe replace this by `f\reduced($item)` so that function gets coverage?

Reply:
Or add a UT for `reduced`, but that might be superfluous.

Reply:
Yep, you're right! 
I'll use `reduced` here.